Sources/NIO/Linux.swift

```diff
@@ -132,5 +135,54 @@ internal enum Linux {
         }
         return fd
     }
+
+    private static func contentsOfFile(path: String) throws -> Substring {
+        let fh = try NIOFileHandle(path: path)
+        defer { try! fh.close() }
+        // linux doesn't properly report /sys/fs/cgroup/* files lengths so we use a reasonable limit
+        var buf = ByteBufferAllocator().buffer(capacity: 1024)
+        try buf.writeWithUnsafeMutableBytes(minimumWritableBytes: buf.capacity) { ptr in
+            let res = try fh.withUnsafeFileDescriptor { fd -> IOResult<ssize_t> in
+                return try Posix.read(descriptor: fd, pointer: ptr.baseAddress!, size: ptr.count)
+            }
+            switch res {
+            case .processed(let n):
+                return n
+            case .wouldBlock:
+                preconditionFailure("read returned EWOULDBLOCK despite a blocking fd")
+            }
+        }
+        return String(buffer: buf).prefix(while: { $0 != "\n" })
+    }
+
+    private static func countCoreIds(cores: Substring) -> Int {
+        let ids = cores.split(separator: "-")
+        guard
+            let first = ids.first.map(String.init).flatMap(Int.init),
+            let last = ids.last.map(String.init).flatMap(Int.init),
+            last > first || !cores.contains("-")
```

Comment:
If we adopt `maxSplits: 1` we can remove these `cores.contains` logic and just use `last >= first`.